web/src/main/java/com/epam/esm/security/SecurityConfig.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+    private final JWTFilter jwtFilter;
+    private final AuthenticationHandlerEntryPoint authenticationHandlerEntryPoint;
+    private final AccessDeniedHandlerEntryPoint accessDeniedHandlerEntryPoint;
+
+    private static final String ADMIN = "ADMIN";
+
+    @Autowired
+    public SecurityConfig(JWTFilter jwtFilter, AuthenticationHandlerEntryPoint authenticationHandlerEntryPoint,
+                          AccessDeniedHandlerEntryPoint accessDeniedHandlerEntryPoint) {
+        this.jwtFilter = jwtFilter;
+        this.authenticationHandlerEntryPoint = authenticationHandlerEntryPoint;
+        this.accessDeniedHandlerEntryPoint = accessDeniedHandlerEntryPoint;
+    }
+
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity
+                .csrf(AbstractHttpConfigurer::disable)
+                .httpBasic(AbstractHttpConfigurer::disable)
+                .sessionManagement(sessionManagement -> sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+                .authorizeHttpRequests(req -> {
+                    req
+                            //All
+                            .requestMatchers("/login/**").permitAll()
+                            .requestMatchers(GET, "/gift_certificates/**").permitAll()
+                            .requestMatchers(POST, "/register", "/auth").permitAll()
+                            //USER
+                            .requestMatchers(POST, "/orders").fullyAuthenticated()
+                            .requestMatchers(GET, "/tags/**", "/users/**", "/orders/**").fullyAuthenticated()
+                            .anyRequest().hasRole(ADMIN);
+                })
+                .exceptionHandling(httpSecurityExceptionHandlingConfigurer -> httpSecurityExceptionHandlingConfigurer.accessDeniedHandler(accessDeniedHandlerEntryPoint).authenticationEntryPoint(authenticationHandlerEntryPoint))
+                .addFilterBefore(jwtFilter, UsernamePasswordAuthenticationFilter.class);
+
+        return httpSecurity.build();
+    }
+
+    @Bean
+    public OAuth2UserService<OAuth2UserRequest, OAuth2User> oauth2UserService() {
```

Comment:
Seems to me, OAuth2 has not been implemented fully. If the App is not using it, please remove unnecessary lines and dependencies